mongostore.go

```diff
@@ -0,0 +1,264 @@
+package quickfix
+
+import (
+	"fmt"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"time"
+
+	"github.com/quickfixgo/quickfix/config"
+)
+
+type mongoStoreFactory struct {
+	settings           *Settings
+	messagesCollection string
+	sessionsCollection string
+}
+
+type mongoStore struct {
+	sessionID          SessionID
+	cache              *memoryStore
+	mongoUrl           string
+	mongoDatabase      string
+	db                 *mgo.Session
+	messagesCollection string
+	sessionsCollection string
+}
+
+// NewMongoStoreFactory returns a mongo-based implementation of MessageStoreFactory
+func NewMongoStoreFactory(settings *Settings) MessageStoreFactory {
+	return NewMongoStoreFactoryPrefixed(settings, "")
+}
+
+// NewMongoStoreFactoryPrefixed returns a mongo-based implementation of MessageStoreFactory, with prefix on collections
+func NewMongoStoreFactoryPrefixed(settings *Settings, collectionsPrefix string) MessageStoreFactory {
+	return mongoStoreFactory{
+		settings:           settings,
+		messagesCollection: collectionsPrefix + "messages",
+		sessionsCollection: collectionsPrefix + "sessions",
+	}
+}
+
+// Create creates a new MongoStore implementation of the MessageStore interface
+func (f mongoStoreFactory) Create(sessionID SessionID) (msgStore MessageStore, err error) {
+	sessionSettings, ok := f.settings.SessionSettings()[sessionID]
+	if !ok {
+		return nil, fmt.Errorf("unknown session: %v", sessionID)
+	}
+	mongoConnectionUrl, err := sessionSettings.Setting(config.MongoStoreConnection)
+	if err != nil {
+		return nil, err
+	}
+	mongoDatabase, err := sessionSettings.Setting(config.MongoStoreDatabase)
+	if err != nil {
+		return nil, err
+	}
+	return newMongoStore(sessionID, mongoConnectionUrl, mongoDatabase, f.messagesCollection, f.sessionsCollection)
+}
+
+func newMongoStore(sessionID SessionID, mongoUrl string, mongoDatabase string, messagesCollection string, sessionsCollection string) (*mongoStore, error) {
+	store := &mongoStore{
+		sessionID:          sessionID,
+		cache:              &memoryStore{},
+		mongoUrl:           mongoUrl,
+		mongoDatabase:      mongoDatabase,
+		messagesCollection: messagesCollection,
+		sessionsCollection: sessionsCollection,
+	}
+	store.cache.Reset()
+
+	if conn, err := mgo.Dial(mongoUrl); err != nil {
+		return nil, err
+	} else {
```

Comment:
Can drop this else.  Can also assign store.db in the if

Reply:
[https://github.com/quickfixgo/quickfix/pull/343/commits/2ae607bc4bb0cbb95140fe6501dba4c364cc5003](
https://github.com/quickfixgo/quickfix/pull/343/commits/2ae607bc4bb0cbb95140fe6501dba4c364cc5003)